src/ansible_navigator/utils/serialize.py

```diff
@@ -26,6 +41,37 @@
 # pylint: enable=unused-import
 
 
+def serialize(
+    content_view: "ContentView",
+    content: "ContentBase",
+    serialization_format: SerializationFormat,
+    file_mode: str = "w",
+    filename: str = None,
```

Comment:
Why build APIs around strings when you can use Pathlib-like interfaces? (with things supporting `__fspath__()`)

Reply:
It was built like this so we could transition the consumers of it across the project without addressing the fact the file is a string elsewhere..........but, I can add a Union here so it doesn't propagate the use of a str moving forward.

Reply:
Add path.

Related: https://github.com/ansible/ansible-navigator/issues/697